### PR TITLE
Basic beats

### PR DESCRIPTION
**Pull request by blacker**

### CODE REVIEW

**Comment by tkell** [comment]
A preliminary overview of this looks sane!  I need to give it more detail when I'm not watching politics though.

File: amen/audio.py [hunk 1]
```diff
@@ -21,8 +22,35 @@ def __init__(self, file_path, convert_to_mono=False, sample_rate=22050):
...
+            y=y_mono, sr=self.sample_rate, trim=False)
+        # convert frames to times
+        beat_times = librosa.frames_to_time(beat_frames, sr=self.sample_rate)
+        # make the list of (start, duration)s that TimingList expects
+        starts_durs = []
+        for i, start in enumerate(beat_times[:-1]):
+            starts_durs.append((start, beat_times[i+1] - start))
+        # now get the last one
+        starts_durs.append((beat_times[-1], self.duration - beat_times[-1]))
+
```
Comment (bmcfee) [inline amen/audio.py]: you can do this bit a bit more succinctly as follows:

```
# pad beat_times to full duration
beat_times = librosa.util.fix_frames(beat_times, x_min=0, x_max=self.duration)
# Generate all inter-beat intervals
starts_durs = [(s, t-s) for (s,t) in zip(beat_times, beat_times[1:])]
```

The above code will pad out to span the range `[0, duration]` and then slice adjacent indices to make intervals.  The benefit of doing it this way is that it handles things like duplicated event times at the boundaries.  If you don't want to add the [0, beat_1] interval, just set `x_min=None`.

File: amen/time.py [hunk 2]
```diff
@@ -6,8 +6,28 @@
...
+    A list of TimeSlices.  
+    """
+
+    def __init__(self, name, timings, audio, unit='s'):
+        # This assumes that we're going to get a list of tuples (start, duration) from librosa,
+        # which may or may not be true.
+        self.name = name
+        for (start, duration) in timings:
+            slice = TimeSlice(start, duration, audio, unit=unit)
+            self.append(slice)
```
Comment (bmcfee) [inline amen/time.py]: `slice` is reserved in python.  Maybe make this `time_slice`?

Comment (tkell) [inline amen/time.py]: :+1: to this

**Comment by tkell** [comment]
Yep, looks good, with Brian's changes!

**Comment by tkell** [comment]
Added a small fix to Ben's `__repr__`.  

(Couldn't find a better way to convert nano-second level timedeltas to seconds, so I just divided.)

**Comment by bmcfee** [comment]
> (Couldn't find a better way to convert nano-second level timedeltas to seconds, so I just divided.)

``timedelta.total_seconds()``?

**Comment by tkell** [comment]
...returns just the seconds, not the total in seconds. = \

**Comment by bmcfee** [comment]
Derp, right.  That [always](https://github.com/marl/jams/blob/master/jams/core.py#L1226-L1229) trips me up.

**Comment by tkell** [comment]
Ah, but `* 1e-9` is so much better.  Ty!

**Comment by tkell** [comment]
..any reasons why we can't merge this, pals?  If I hear nothing, I'll do so sometime this week.

**Comment by tkell** [comment]
Made tweaks as suggested by @bmcfee.  Will fix the conflicts and merge tomorrow, unless there are fires and portents.

**Pull request merged**